src/dao/api.dao.ts

```diff
@@ -335,6 +345,73 @@ export class NodeDao {
     }
   };
 
+  async getFeedEntriesV3(limit: number, cursor: string | null, query: string | null) {
+    try {
+      query = query ? `%${query}%` : null;
+      cursor = cursor ? Buffer.from(cursor, "base64").toString() : null;
+
+      const rawQ = Prisma.sql`
+        WITH RankedClaims AS (
+          SELECT
+            n.name AS name,
+            n."nodeUri" AS link,
+            c.id AS claim_id,
+            c.statement AS statement,
+            c.stars AS stars,
+            c."effectiveDate" AS effective_date,
+            ROW_NUMBER() OVER (PARTITION BY c.id) AS row_num,
+            CONCAT(COALESCE(to_char(c."effectiveDate", 'YYYYMMDDHH24MISS'), ''), c.id::TEXT) AS cursor
+          FROM "Claim" c
+          INNER JOIN "Edge" AS e ON c.id = e."claimId"
+          INNER JOIN "Node" AS n ON e."startNodeId" = n.id
+          WHERE
+            n."entType" != 'CLAIM'
+            AND e.label != 'source'
+            AND c."effectiveDate" IS NOT NULL
+            AND c.statement IS NOT NULL
+            AND n.name IS NOT NULL
+            AND n.name != ''
+            AND (
+              c.subject ILIKE COALESCE(${query}, '%') OR
+              c.statement ILIKE COALESCE(${query}, '%') OR
```

Comment:
this is going to be slow, we might want to try without anything that does not have an index

Reply:
it is ok for the query i think but, we could also query first for name or url and fallback to statement only if we don't have those kind of matches.  anyway its ok, but i think our other problems are really separate from this

* initial feed is slow (could precache, does not require search, why is it slow if its just most recent?)

* when going to graph we should NOT do query but search on id exactly only

anyway this is ok, its fine for a try, but just mentioning those are probably more pressing problems

Reply:
When I tested it, the diff wasn't that big. We can improve it more by text indexing the fields we need to search with, and as you said. rely on fields more suitable for searching. I've removed some fields that need to join a lot of tables just to search in them without really showing any data that is used to search. That was the huge gain came from, removing unused joins. Now the request takes less than 1s, ofc that's not the best but at least it's a good step.

After using this endpoint we'll have a better idea on what is the bottleneck. The we can improve with the new information

Reply:
ok!  ha i just edited same code, sorry about that!  but yes lets get it in that's fine, if this is working i can approve, don't wait for me in any case